plutus-tx/src/PlutusTx/List.hs

```diff
@@ -101,7 +103,8 @@ findIndex p l = listToMaybe (findIndices p l)
 --
 infixl 9 !!
 (!!) :: [a] -> Integer -> a
-[]       !! _ = Builtins.error ()
+_        !! n | n < 0 = traceError "PlutusTx.List.!!: negative index"
```

Comment:
This is using the Haskell prelude `<` instead of the Plutus Tx one.

Reply:
Oh, good catch! Thanks!

Reply:
It forced to move `Min(..)` and `Max(..)` to `PlutusTx.Semigroup`.

Reply:
Fine. Alternatively you could use the builtin directly, but either way is good.